types/rlp_unmarshal.go

```diff
@@ -312,75 +312,120 @@ func (t *Transaction) UnmarshalRLPFrom(p *fastrlp.Parser, v *fastrlp.Value) erro
 
 	p.Hash(t.Hash[:0], v)
 
+	// Setup defaults
+	t.GasPrice = new(big.Int)
+	t.Value = new(big.Int)
+	t.V = new(big.Int)
+	t.R = new(big.Int)
+	t.S = new(big.Int)
+	t.From = ZeroAddress
+
 	// nonce
 	if t.Nonce, err = elems[0].GetUint64(); err != nil {
 		return err
 	}
+
 	// gasPrice
-	t.GasPrice = new(big.Int)
-	if err := elems[1].GetBigInt(t.GasPrice); err != nil {
+	if err = elems[1].GetBigInt(t.GasPrice); err != nil {
 		return err
 	}
+
 	// gas
 	if t.Gas, err = elems[2].GetUint64(); err != nil {
 		return err
 	}
+
 	// to
 	if vv, _ := v.Get(3).Bytes(); len(vv) == 20 {
 		// address
 		addr := BytesToAddress(vv)
 		t.To = &addr
-	} else {
-		// reset To
-		t.To = nil
 	}
+
 	// value
-	t.Value = new(big.Int)
-	if err := elems[4].GetBigInt(t.Value); err != nil {
+	if err = elems[4].GetBigInt(t.Value); err != nil {
 		return err
 	}
+
 	// input
 	if t.Input, err = elems[5].GetBytes(t.Input[:0]); err != nil {
 		return err
 	}
 
 	// V
-	t.V = new(big.Int)
 	if err = elems[6].GetBigInt(t.V); err != nil {
 		return err
 	}
 
 	// R
-	t.R = new(big.Int)
 	if err = elems[7].GetBigInt(t.R); err != nil {
 		return err
 	}
+
 	// S
-	t.S = new(big.Int)
 	if err = elems[8].GetBigInt(t.S); err != nil {
 		return err
 	}
 
-	if len(elems) >= 10 {
+	if len(elems) > 9 {
 		// Type
-		if t.Type, err = ReadRlpTxType(elems[9]); err != nil {
+		if t.Type, err = readRlpTxType(elems[9]); err != nil {
```

Comment:
I do not think this is correct. The type of the transaction is not on element 9 as far as I know but at the beginning of the transaction.

Reply:
It is correct, because Roman changed it in marshalling logic, to be at index=9, so that we are backward compatible:
https://github.com/0xPolygon/polygon-edge/pull/1111/files#diff-7f00e74f75ee3a971d40d6f05e54168503eb91a38717d7ed211dafab18ef478cR194

Reply:
By backward compatible, we meant with the previous polybft versions which have this enabled? Ok, what I mean is not that but that there is an official specification to encode an Ethereum transaction with a type (https://eips.ethereum.org/EIPS/eip-2718). I do not think this encoding (or for that matter) or the previous one follows that format.

Reply:
> By backward compatible, we meant with the previous polybft versions which have this enabled?

By backward compatibility I was thinking about transactions which were created and are present in blocks before we've introduced this change and `Type` field in the first place (doesn't matter is it IBFT or PolyBFT). I would imagine we would encounter a use case where some end-users would like to still run IBFT and keep running with "old" transaction data format. Am I correct there?

Reply:
@ferranbt i didn't change the order of elements in rlp marshaling. Just added gas fee cap and gas tip cap to the end. I guess it would be better to update these things within a separate PR just to keep things more accurate then. Wdyt? 
And yes, I though the same as @Stefan-Ethernal regarding backward compatibility. Does that mean that we can change the order then @ferranbt ?